### PR TITLE
Improve the formatting of lists in doc comments

### PR DESCRIPTION
**Pull request by diml**
This implements #477.

### CODE REVIEW

File: src/Fmt_odoc.ml [hunk 1]
```diff
@@ -85,7 +89,29 @@ and fmt_text_elt = function
...
-and fmt_item txt = hovbox 0 (wrap "{- " "}" (fmt_text txt))
+and fmt_list (kind : List_kind.t) l =
+  let light_syntax =
+    let line_start = fmt (match kind with List -> "- " | Enum -> "+ ") in
+    let fmt_item txt = hovbox 2 (line_start $ fmt_text txt) in
+    vbox 0 (list l "@," fmt_item)
+  in
+  let heavy_syntax =
+    let fmt_item txt = hovbox 0 (wrap "{- " "}" (fmt_text txt)) in
+    let start : (_, _, _) format =
```
Comment (jberdine) [inline src/Fmt_odoc.ml]: FMI: is it important that this is a 3-arg format, or would `_ format` be good enough?

File: src/Fmt_odoc.ml [hunk 2]
```diff
@@ -85,7 +89,29 @@ and fmt_text_elt = function
...
+      match kind with List -> "{ul@," | Enum -> "{ol@,"
+    in
+    vbox 1 (wrap start "}" (list l "@," fmt_item))
+  in
+  let print_and_parse fmt =
+    let str = Format.asprintf "\n%t" fmt in
+    Octavius.parse (Lexing.from_string str)
+  in
+  if
+    Poly.equal (print_and_parse light_syntax) (print_and_parse heavy_syntax)
```
Comment (jberdine) [inline src/Fmt_odoc.ml]: nice

File: test/passing/doc_comments.mli [hunk 3]
```diff
@@ -127,3 +127,41 @@ val x : x
         s |> String.split_lines |> List.map ~f:String.strip
         |> fun s -> list s "@," Fmt.str
     ]} *)
+
+(** Lists:
+
+    list with short lines:
+
```
Comment (jberdine) [inline test/passing/doc_comments.mli]: Is this open line necessary, or just desired?

File: src/Fmt_odoc.ml [hunk 4]
```diff
@@ -37,6 +37,10 @@ let fmt_ref_kind = function
   | RK_link -> str ":"
   | RK_custom s -> str s $ str ":"
 
+module List_kind = struct
+  type t = List | Enum
+end
```
Comment (jberdine) [inline src/Fmt_odoc.ml]: Nothing significant, but I'm not sure what the reasoning is behind adding a module vs just using a couple polymorphic variant tags?

File: src/Fmt_odoc.ml (hunk 1 above)
Comment (diml) [reply src/Fmt_odoc.ml]: Indeed, I updated the code

File: src/Fmt_odoc.ml (hunk 4 above)
Comment (diml) [reply src/Fmt_odoc.ml]: It's just a habit I have, I prefer to only use polymorphic variants when I need one of the specific features of polymorphic variants. I find that with regular variants you get errors closer to the actual location of the problem and nicer error messages.

I updated the code to use polymorphic variants instead.

**Comment by diml** [comment]
Thanks, I updated the code

**Pull request merged**